File: internal/content/testdata/bench/javascript/nc-group-anagrams.js
function group_anagrams(strs) {
    const groups = new Map();
    for (const s of strs) {
        const key = s.split('').sort().join('');
        if (!groups.has(key)) groups.set(key, []);
        groups.get(key).push(s);
    }
    const res = Array.from(groups.values()).map(g => g.slice().sort());
    res.sort((a, b) => {
        const minLen = Math.min(a.length, b.length);
        for (let i = 0; i < minLen; i++) {
            if (a[i] < b[i]) return -1;
            if (a[i] > b[i]) return 1;
        }
        return a.length - b.length;
    });
    return res;
}

File: internal/content/testdata/bench/java/nc-group-anagrams.java
import java.util.*;

class Solution {
    public String[][] group_anagrams(String[] strs) {
        Map<String, List<String>> groups = new HashMap<>();
        for (String s : strs) {
            char[] arr = s.toCharArray();
            Arrays.sort(arr);
            String key = new String(arr);
            groups.computeIfAbsent(key, k -> new ArrayList<>()).add(s);
        }
        List<List<String>> res = new ArrayList<>();
        for (List<String> g : groups.values()) {
            Collections.sort(g);
            res.add(g);
        }
        res.sort((a, b) -> {
            for (int i = 0; i < Math.min(a.size(), b.size()); i++) {
                int cmp = a.get(i).compareTo(b.get(i));
                if (cmp != 0) return cmp;
            }
            return a.size() - b.size();
        });
        String[][] out = new String[res.size()][];
        for (int i = 0; i < res.size(); i++) {
            out[i] = res.get(i).toArray(new String[0]);
        }
        return out;
    }
}
